test/mocha.js

```diff
@@ -236,4 +270,53 @@ describe('Switcher test suite', () => {
       });
     });
   });
+
```

Comment:
Need some tests here that cover those edge cases that I previously mentioned, just to make sure that we're not screwing up anybody's file format.